### PR TITLE
Update qt to version 6, python to version 3

### PR DESCRIPTION
**Pull request by biagas**
Updated Find modules and usage in CMakeLists.txt 
Updated LC config-site file.

I successfully compiled and ran tests on Windows and LC (pascal).
silex runs on both systems.

### CODE REVIEW

File: CMake/SiloFindPython.cmake [hunk 1]
```diff
@@ -58,26 +58,31 @@
...
+    set(Python3_ROOT_DIR ${SILO_PYTHON_DIR})
+    if(WIN32)
+        set(Python3_EXECUTABLE ${SILO_PYTHON_DIR}/python.exe)
+    else()
+        set(Python3_EXECUTABLE ${SILO_PYTHON_DIR}/bin/python3)
+    endif()
 endif()
 
-find_package(Python COMPONENTS Interpreter Development)
+find_package(Python3 COMPONENTS Interpreter Development)
```
Comment (markcmiller86) [inline CMake/SiloFindPython.cmake:71]: Does this mean we now require python3? I'd prefer not to have to *require* it if we can avoid that.

**Review by markcmiller86 — COMMENTED** [review]
I just had one question.

File: CMake/SiloFindPython.cmake (hunk 1 above)
Comment (biagas) [reply CMake/SiloFindPython.cmake:71]: Yes. I thought since Python 2 is no longer supported, that we should move to the new version for new releases.

Comment (markcmiller86) [reply CMake/SiloFindPython.cmake:71]: Yeah, Python 2 is no longer supported but a lot of computing centers still have it as the default and a lot of our customers I think still use it. @cyrus do you have thoughts on this?

Comment (markcmiller86) [reply CMake/SiloFindPython.cmake:71]: I think Python 2 is the default in a lot of computing centers and I think a lot of our customers still use Python 2. @cyrush do you have thoughts on this?

Comment (biagas) [reply CMake/SiloFindPython.cmake:71]: I can remove the python3 changes if you want, no problem.

Comment (markcmiller86) [reply CMake/SiloFindPython.cmake:71]: Lets leave at python2 for now. With autotools, I know how to override to get python3 instead. I dunno how to do that with CMake.

**Review by markcmiller86 — APPROVED** [review]

File: CMake/SiloFindPython.cmake (hunk 1 above)
Comment (biagas) [reply CMake/SiloFindPython.cmake]: I reverted the changes that forced python3. Kept mods that help CMake find the python specified by SILO_PYTHON_DIR if provided.  Is preferring python3 in this case okay?

**Review by markcmiller86 — APPROVED** [review]

**Pull request merged**